### PR TITLE
hasOnlyOneElementSatisfying() should be specified in ObjectEnumerableAssert

### LINKED ISSUE #1302: hasOnlyOneElementSatisfying() should be specified in ObjectEnumerableAssert
#### Summary

I noticed that `AbstractIterableAssert.hasOnlyOneElementSatisfying()` is not only one of the only assertions not returning SELF (see #1301), but that it is also one of the only methods not specified in the interface `ObjectEnumerableAssert`. For consistency perhaps it should be specified in the interface as well.

### PR DESCRIPTION
**Pull request by Mef1STo**
#### Check List:
* Fixes #1302
* Unit tests : YES
* Javadoc with a code example (API only) : YES

### CODE REVIEW

**Comment by joel-costigliola** [comment]
Thanks, I will review this PR shortly !

File: src/test/java/org/assertj/core/api/atomic/referencearray/AtomicReferenceArrayAssert_hasOnlyOneElementSatisfying_Test.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  @Override
+  protected AtomicReferenceArrayAssert<Object> invoke_api_method() {
+    return assertions.hasOnlyOneElementSatisfying(consumer);
+  }
+
+  @Override
+  protected void verify_internal_effects() {
+    verify(iterables).assertHasOnlyOneElementSatisfying(info(), newArrayList(internalArray()), consumer);
```
[the remark this answers is not in the document]
Comment (Mef1STo) [reply src/test/java/org/assertj/core/api/atomic/referencearray/AtomicReferenceArrayAssert_hasOnlyOneElementSatisfying_Test.java]: changed

File: src/main/java/org/assertj/core/api/AtomicReferenceArrayAssert.java [hunk 2]
```diff
@@ -202,6 +202,13 @@ public void isEmpty() {
     return myself;
   }
 
+  /** {@inheritDoc} */
```
[the remark this answers is not in the document]
Comment (Mef1STo) [reply src/main/java/org/assertj/core/api/AtomicReferenceArrayAssert.java]: Wrote, but don't sure that I correct understand what you mean about code example section

File: src/main/java/org/assertj/core/api/AbstractObjectArrayAssert.java [hunk 3]
```diff
@@ -326,59 +326,12 @@ public SELF containsOnlyNulls() {
...
-   * assertThat(jedis).hasOnlyOneElementSatisfying(yoda -&gt; assertThat(yoda.getName()).startsWith("Yo"));</code></pre>
-   *
-   * @param elementAssertions the assertions to perform on the unique element.
-   * @throws AssertionError if the array does not have a unique element.
-   * @throws AssertionError if the array's unique element does not satifies the given assertions.
-   *
-   * @since 3.5.0
-   */
-  public void hasOnlyOneElementSatisfying(Consumer<ELEMENT> elementAssertions) {
+  /** {@inheritDoc} */
```
[the remark this answers is not in the document]
Comment (Mef1STo) [reply src/main/java/org/assertj/core/api/AbstractObjectArrayAssert.java]: Returned

File: src/main/java/org/assertj/core/api/AtomicReferenceArrayAssert.java [hunk 4]
```diff
@@ -202,6 +202,66 @@ public void isEmpty() {
...
+   * jedis = new AtomicReferenceArray&lt;&gt;(new Jedi[]{
+   *   new Jedi("Yoda", "red"),
+   *   new Jedi("Luke", "green")
+   * });
+   * assertThat(jedis).hasOnlyOneElementSatisfying(yoda -&gt; assertThat(yoda.getName()).startsWith("Yo"));</code></pre>
+   *
+   * @param elementAssertions the assertions to perform on the unique element.
+   * @throws AssertionError if the {@link Iterable} does not have a unique element.
+   * @throws AssertionError if the {@link Iterable}'s unique element does not satisfies the given assertions.
+   * @since 3.12.0
```
Comment (Mef1STo) [inline src/main/java/org/assertj/core/api/AtomicReferenceArrayAssert.java]: I'm not sure that I set the correct version here

Comment (joel-costigliola) [reply src/main/java/org/assertj/core/api/AtomicReferenceArrayAssert.java]: nope, that's all good!

File: src/main/java/org/assertj/core/api/AbstractObjectArrayAssert.java [hunk 5]
```diff
@@ -372,13 +372,14 @@ public SELF containsOnlyNulls() {
...
    * @param elementAssertions the assertions to perform on the unique element.
    * @throws AssertionError if the array does not have a unique element.
-   * @throws AssertionError if the array's unique element does not satifies the given assertions.
-   *
+   * @throws AssertionError if the array's unique element does not satisfies the given assertions.
    * @since 3.5.0
    */
-  public void hasOnlyOneElementSatisfying(Consumer<ELEMENT> elementAssertions) {
+  @Override
+  public SELF hasOnlyOneElementSatisfying(Consumer<ELEMENT> elementAssertions) {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractObjectArrayAssert.java]: Change to `Consumer<ELEMENT>` to `Consumer<? super ELEMENT>`

File: src/main/java/org/assertj/core/api/AtomicReferenceArrayAssert.java [hunk 6]
```diff
@@ -202,6 +202,66 @@ public void isEmpty() {
...
+   */
+  @Override
+  public AtomicReferenceArrayAssert<T> hasOnlyOneElementSatisfying(Consumer<T> elementAssertions) {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AtomicReferenceArrayAssert.java]: Change to `Consumer<T>` to `Consumer<? super T>`

File: src/main/java/org/assertj/core/api/ObjectEnumerableAssert.java [hunk 7]
```diff
@@ -886,6 +886,57 @@
...
+   * // even if the assertion is correct, there are too many jedis !
+   * jedis.add(new Jedi("Luke", "green"));
+   * assertThat(jedis).hasOnlyOneElementSatisfying(yoda -&gt; assertThat(yoda.getName()).startsWith("Yo"));</code></pre>
+   *
+   * @param elementAssertions the assertions to perform on the unique element.
+   * @throws AssertionError if the {@link Iterable} does not have a unique element.
+   * @throws AssertionError if the {@link Iterable}'s unique element does not satisfies the given assertions.
+   * @since 3.5.0
+   */
+  SELF hasOnlyOneElementSatisfying(Consumer<ELEMENT> elementAssertions);
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/ObjectEnumerableAssert.java]: Change to `Consumer<ELEMENT>` to `Consumer<? super ELEMENT>`

File: src/main/java/org/assertj/core/api/AtomicReferenceArrayAssert.java (hunk 4 above)
Comment (joel-costigliola) [reply src/main/java/org/assertj/core/api/AtomicReferenceArrayAssert.java]: This is correct

**Comment by joel-costigliola** [comment]
@Mef1STo sorry for the late feedback, a few more changes and that will be good!

**Comment by Mef1STo** [comment]
OK, I'll fix it soon)

**Comment by joel-costigliola** [comment]
Integrated thanks @Mef1STo !

**Pull request closed without merging**